### PR TITLE
Link constraints management

### PR DESCRIPTION
**Pull request by hugo-antoine-rtei**
Contains the logic for setting the NTC values to 0 during the first step of the adequacy patch. See [our internal PR](https://github.com/rte-i/Antares_Simulator/pull/8) for more info.

### CODE REVIEW

File: src/libs/antares/study/area/area.cpp [hunk 1]
```diff
@@ -53,6 +53,7 @@ void Area::internalInitialize()
 Area::Area() :
  index((uint)(-1)),
  enabled(true),
+ adequacyPatchMode(Data::adqmPhysicalAreaOutsideAdqPatch),
```
Comment (flomnes) [inline src/libs/antares/study/area/area.cpp:56]: I suggest in-class initialization. Also,
```suggestion
 adequacyPatchMode(Data::AdequacyPatch::PhysicalAreaOutside),
```

File: src/libs/antares/study/area/area.h [hunk 2]
```diff
@@ -225,6 +225,8 @@ class Area final : private Yuni::NonCopyable<Area>
     uint index;
     //! Enabled
     bool enabled;
+    //! Use adequacy patch for this area
+    Data::AdequacyPatchMode adequacyPatchMode;
```
Comment (flomnes) [inline src/libs/antares/study/area/area.h:229]:
```suggestion
    Data::AdequacyPatch::Mode adequacyPatchMode;
```

File: src/libs/antares/study/area/list.cpp [hunk 3]
```diff
@@ -323,6 +331,30 @@ bool saveAreaOptimisationIniFile(const Area& area, const Clob& buffer)
...
+    case Data::adqmPhysicalAreaOutsideAdqPatch:
+        value = 1;
+        break;
+    case Data::adqmPhysicalAreaInsideAdqPatch:
+        value = 2;
+        break;
+    default:
+        value = 1; //default adqmPhysicalAreaOutsideAdqPatch
+        break;
+    }
```
Comment (flomnes) [inline src/libs/antares/study/area/list.cpp:353]: I suggest extracting this conversion into a dedicated function. You might need it elsewhere.

File: src/libs/antares/study/area/list.cpp [hunk 4]
```diff
@@ -1037,6 +1069,44 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
...
+                    case 1:
+                        area.adequacyPatchMode = Data::adqmPhysicalAreaOutsideAdqPatch;
+                        break;
+                    case 2:
+                        area.adequacyPatchMode = Data::adqmPhysicalAreaInsideAdqPatch;
+                        break;
+                    default:
+                        area.adequacyPatchMode = Data::adqmPhysicalAreaOutsideAdqPatch;
+                        break;
+                    }
```
Comment (flomnes) [inline src/libs/antares/study/area/list.cpp:1103]: I suggest extraction this conversion function.

File: src/libs/antares/study/fwd.h [hunk 5]
```diff
@@ -116,6 +116,36 @@ enum StudyMode
 
 }; // enum StudyMode
 
+/*!
+** \brief Types of Adequacy patch mode
+*/
+enum AdequacyPatchMode
```
Comment (flomnes) [inline src/libs/antares/study/fwd.h:122]:
```
namespace AdequacyPatch {
//...
}
```

File: src/libs/antares/study/fwd.h [hunk 6]
```diff
@@ -116,6 +116,36 @@ enum StudyMode
...
+    adqmPhysicalAreaOutsideAdqPatch,
+    //! Physical Area inside the adq-patch
+    adqmPhysicalAreaInsideAdqPatch
+
+}; // enum AdequacyPatchMode
+
+/*!
+** \brief Setting NTC for Adequacy patch first step
+*/
+enum NTC
```
Comment (flomnes) [inline src/libs/antares/study/fwd.h:136]: Inside namespace `AdequacyPatch`

File: src/libs/antares/study/fwd.h [hunk 7]
```diff
@@ -116,6 +116,36 @@ enum StudyMode
...
+{
+    //! Leave NTC local values
+    leaveLocalValues = 0,
+    //! Set NTC to zero
+    setToZero,
+    //! set only start->end NTC to zero
+    setStartEndToZero,
+    //! set only end->start NTC to zero
+    setEndStartToZero
```
Comment (flomnes) [inline src/libs/antares/study/fwd.h:145]: We use `from` and `to` for links.

File: src/ui/simulator/windows/inspector/frame.cpp [hunk 8]
```diff
@@ -785,6 +790,9 @@ void Frame::apply(const InspectorData::Ptr& data)
             pPGAreaName->SetValueFromString(wxStringFromUTF8((*(data->areas.begin()))->name));
         // Area color
         Accumulator<PAreaColor>::Apply(pPGAreaColor, data->areas);
+        // Adequacy patch
+        if (!multiple)
```
Comment (flomnes) [inline src/ui/simulator/windows/inspector/frame.cpp:794]: Why exclude `multiple` ?

Comment (hugo-antoine-rtei) [reply src/ui/simulator/windows/inspector/frame.cpp:794]: We copied what happened around it (like for the Name of the Area & Area position), I'm not sure there was much more thought put into it.

**Pull request closed without merging**